debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -73,6 +78,20 @@
 
     private Duration retriableRestartWait;
 
+    private final ElapsedTimeStrategy pollOutputDelay;
+    private final Clock clock = Clock.system();
+    private long recordCounter = 0L;
+    private Instant previousOutputInstant;
```

Comment:
Does the last two need to be volatile (concurrent access)?

Reply:
No, they don't as in both cases they are accessed only on producer side - in case of queue, or consumer side in case for task polling message